### PR TITLE
Reintroduce boringcrypto images

### PR DESCRIPTION
**Pull request by rfratto**
This PR introduces boringcrypto/cngcrypto Docker images to mirror boringcrypto/cngcrypto support in Grafana Agent.

Related to #64.

I'll follow up with another PR for distributing boringcrypto binaries after this is merged.

### CODE REVIEW

File: docs/sources/get-started/install/docker.md [hunk 1]
```diff
@@ -50,6 +50,19 @@ Make sure you pass `--server.http.listen-addr=0.0.0.0:12345` as an argument as s
 If you don't pass this argument, the [debugging UI][UI] won't be available outside of the Docker container.
 {{< /admonition >}}
 
+### Boringcrypto images
+
+> **Note**: `boringcrypto` support is in _Public preview_ and is only available
+> on AMD64 and ARM64 platforms.
```
Comment (clayton-cornell) [inline docs/sources/get-started/install/docker.md:56]:
```suggestion
{{< admonition type="note" >}}
`boringcrypto` support is in _Public preview_ and is only available on AMD64 and ARM64 platforms.
{{< /admonition >}}
```

File: docs/sources/get-started/install/docker.md [hunk 2]
```diff
@@ -75,6 +88,20 @@ Make sure you pass `--server.http.listen-addr=0.0.0.0:12345` as an argument as s
...
+
+`cngcrypto` images are published with every release starting with version
+1.1:
+
+* The latest `cngcrypto` image is published as `grafana/alloy:nanoserver-1809-cngcrypto`.
+* A specific version of the `cngcrypto` image is published as
+  `grafana/alloy:<VERSION>-nanoserver-1809-cngcrypto`, such as
+  `grafana/alloy:v1.1.0-nanoserver-1809-cngcrypto`.
+
+
```
Comment (clayton-cornell) [inline docs/sources/get-started/install/docker.md:104]:
```suggestion
  `grafana/alloy:v1.1.0-nanoserver-1809-cngcrypto`.

```

Extra CR/LF

File: docs/sources/get-started/install/docker.md [hunk 3]
```diff
@@ -75,6 +88,20 @@ Make sure you pass `--server.http.listen-addr=0.0.0.0:12345` as an argument as s
 If you don't pass this argument, the [debugging UI][UI] won't be available outside of the Docker container.
 {{< /admonition >}}
 
+### Cngcrypto images
+
+> **Note**: `cngcrypto` support is in _Public preview_ and is only available
+> on AMD64 platforms.
```
Comment (clayton-cornell) [inline docs/sources/get-started/install/docker.md:94]:
```suggestion
{{< admonition type="note" >}}
`cngcrypto` support is in _Public preview_ and is only available on AMD64 platforms.
{{< /admonition >}}
```

File: .drone/drone.yml [hunk 4]
```diff
@@ -507,8 +627,43 @@ volumes:
     path: //./pipe/docker_engine/
   name: docker
 ---
+kind: pipeline
+name: Publish Windows alloy-boringcrypto container
```
Comment (mattdurham) [inline .drone/drone.yml:631]: Should we name the windows ones cngcrypto instead of boring, since that matches our naming?

**Review by mattdurham — APPROVED** [review]
LGTM with one small comment

File: .drone/drone.yml (hunk 4 above)
Comment (rfratto) [reply .drone/drone.yml:631]: Yeah this is a bug actually, it should've been `alloy-cngcrypto` here. Thanks for catching :)

**Pull request merged**